contrib/ruby/test/client_test.rb

```diff
@@ -264,6 +264,55 @@ def test_read_timeout
     ensure_closed client
   end
 
+  def test_adjustable_read_timeout
+    client = new_tcp_client(read_timeout: 5)
+    assert client.query("SELECT SLEEP(0.2)");
+    client.read_timeout = 0.1
+    assert_equal 0.1, client.read_timeout
+    assert_raises Errno::ETIMEDOUT do
+      client.query("SELECT SLEEP(1)")
+    end
+  ensure
+    ensure_closed client
+  end
+
+  def test_read_timeout_closed_connection
+    client = new_tcp_client(read_timeout: 5)
+    client.close
+    ensure_closed client
+
+    assert_raises IOError do
+      client.read_timeout
+    end
```

Comment:
So this is a bit weird, but seemed the most logical.

Alternatively I could return a fallback value, or I could keep a copy of the timeout value in the client object, and return that.